### PR TITLE
`SyliusTemplateEvents` introduced on the `Refund` view

### PR DESCRIPTION
**Pull request by Rafikooo**
| Q               | A
| --------------- | -----
| Branch?         | 1.4
| Bug fix?        | no
| New feature?    | no

These events were introduced to ease `Plus` installation and to remove redundant templates

### CODE REVIEW

**Review by coldic3 — APPROVED** [review]

**Review by jakubtobiasz — COMMENTED** [review]
Why some files are empty?

File: src/Resources/config/app/events.yaml [hunk 1]
```diff
@@ -55,3 +55,69 @@ sylius_ui:
                     template: '@SyliusRefundPlugin/Order/Admin/CreditMemo/_downloadButton.html.twig'
                     enabled: '%sylius_refund.pdf_generator.enabled%'
                     priority: 10
+
+        sylius_refund.order.admin.refund.body:
```
Comment (GSadee) [inline src/Resources/config/app/events.yaml]:
```suggestion
        sylius_refund.admin.order.refund.body:
```

Comment (GSadee) [reply src/Resources/config/app/events.yaml]: And in other events below 👇🏻

File: src/Resources/views/Order/Admin/Refund/Body/_bulkButtons.html.twig [hunk 2]
```diff
@@ -0,0 +1,7 @@
+<div class="ui right aligned grid">
+    <div class="sixteen wide column">
+        <button data-refund-clear type="button" class="ui button" {{ disableButton }}>{{ 'sylius_refund.ui.clear_refunds'|trans }}</button>
+        <button data-refund-all type="button" class="ui button primary" {{ disableButton }}>{{ 'sylius_refund.ui.refund_all'|trans }}</button>
```
Comment (GSadee) [inline src/Resources/views/Order/Admin/Refund/Body/_bulkButtons.html.twig]: I'm not sure if this makes sense and if it's not overengineering, but there could be one event here and these buttons would be added by blocks

File: src/Resources/views/Order/Admin/Refund/Body/_form.html.twig [hunk 3]
```diff
@@ -0,0 +1,20 @@
+<form action="{{ path('sylius_refund_refund_units', {'orderNumber': app.request.attributes.get('orderNumber')}) }}" method="post">
+    <input type="hidden" name="_csrf_token" value="{{ csrf_token(app.request.attributes.get('orderNumber')) }}" />
+    <table id="refunds" class="ui compact celled table">
+        <thead>
+        <tr>
+            {{ sylius_template_event('sylius_refund.order.admin.refund.table.header', _context) }}
+        </tr>
+        </thead>
+        <tbody>
+        {{ sylius_template_event('sylius_refund.order.admin.refund.table.row', _context) }}
```
Comment (GSadee) [inline src/Resources/views/Order/Admin/Refund/Body/_form.html.twig]:
```suggestion
        {{ sylius_template_event('sylius_refund.order.admin.refund.table.body', _context) }}
```
or
```suggestion
        {{ sylius_template_event('sylius_refund.order.admin.refund.table.content', _context) }}
```

File: src/Resources/views/Order/Admin/Refund/Body/_form.html.twig [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+        </tbody>
+    </table>
+    {% include '@SyliusRefundPlugin/_paymentMethod.html.twig' %}
```
Comment (GSadee) [inline src/Resources/views/Order/Admin/Refund/Body/_form.html.twig]: In order not to have to include this template here, you could create a single template event `...form.content` with blocks for table, payment method and footer

File: src/Resources/views/Order/Admin/Refund/Body/_form.html.twig [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+    <div class="ui stackable grid">
+        <div class="two column row">
+            {{ sylius_template_event('sylius_refund.admin.order.refund.footer', _context) }}
```
Comment (GSadee) [inline src/Resources/views/Order/Admin/Refund/Body/_form.html.twig]: According to the comment above:
```suggestion
            {{ sylius_template_event('sylius_refund.admin.order.refund.footer.content', _context) }}
```

**Review by GSadee — COMMENTED** [review]
Changing the name of a template is a BC break, the same is about removing the existing templates, so you need to revert these changes, ref. https://docs.sylius.com/en/1.12/book/organization/backward-compatibility-promise.html?highlight=backward#templates

**Comment by Rafikooo** [comment]
> Changing the name of a template is a BC break, the same is about removing the existing templates, so you need to revert these changes, ref. https://docs.sylius.com/en/1.12/book/organization/backward-compatibility-promise.html?highlight=backward#templates

I forget to check that. All deleted/renamed/moved files reverted 🖖

File: src/Resources/views/Order/Admin/Refund/Body/_bulkButtons.html.twig (hunk 2 above)
Comment (Rafikooo) [reply src/Resources/views/Order/Admin/Refund/Body/_bulkButtons.html.twig]: IMO good point, it opens possibility to more buttons 😄

File: src/Resources/views/Order/Admin/Refund/Body/_form.html.twig (hunk 4 above)
Comment (Rafikooo) [reply src/Resources/views/Order/Admin/Refund/Body/_form.html.twig]: I've decided not to use `content` suffix as it's quite generic and could be added to every event. In fact, each block could be treated as some kind of content. Wdyt?

File: src/Resources/views/Order/Admin/Refund/Body/_form.html.twig (hunk 5 above)
Comment (Rafikooo) [reply src/Resources/views/Order/Admin/Refund/Body/_form.html.twig]: Commented above

File: src/Resources/views/Order/Admin/Refund/Body/_form.html.twig (hunk 4 above)
Comment (GSadee) [reply src/Resources/views/Order/Admin/Refund/Body/_form.html.twig]: It makes sense, however, we've used this keyword in template events in Sylius, but for now, there is no rule or any convention about the naming of events

**Review by GSadee — APPROVED** [review]

**Comment by GSadee** [comment]
Thank you, Rafał! :1st_place_medal:

**Pull request merged**